### DISCUSSION TITLE
A few issues using multiple versions of VAPOR (3.9.1 and 3.2.9) on Mac and LINUX

### DISCUSSION THREAD

**Discussion by jinwoongyoo** [NCAR/VAPOR#3516] [Q&A]
Hi, 
I have produced my VDC of my WRF simulations using VAPOR v.3.2.0 on a cluster computer where I couldn't set up VAPOR 3.9.1 due to library version limitation of the cluster.
Then, I downloaded the VDC files to my MacBook Pro to generate VAPOR image files using VAPOR version 3.9.1.
While 2D and 3D Volume rendering went through fine, I could not generate 3D flow, stream line, barb images using U, V, and W.
When tried, I only get very "vertical sticks" for each point of seeds.
I am wondering if this behaviour is results of my mixed versions of VAPOR: one for vdc files and another for actual rendering?
Or can this be a bug of the version 3.9.1?
Previously with VAPOR version 2.x, I was able to generate 3D flow, stream line, barb images using U, V, and W without any issue.

Also, I could not generate 2D or 3D images for soil moisture (SMOIS) of WRF simulation due to like a file coordination issue(?).
I wonder if any of you had tested or plotted any land surface variables of WRF outputs using VAPOR 3.9.1?

Finally, I want to calculate the difference of a variable (aka QVAPOR) from two different vdc file.
I tried to use python tool on the VAPOR 3.9.1 but it seems that I cannot open two different vdc datasets in the tool.
Is there any way I can do the calculation? 

Any comments will be highly appreciated.
Thank you.
Jinwoong

**Comment by sgpearse** [comment]
Hi Jinwoong, that is a lot of questions and I'm glad to see your enthusiasm.  The best place to start is to provide a sample file so we can try to replicate what you're seeing, especially since you are using multiple versions of VAPOR.  There are probably some bugs we need to link to your report but we need to reproduce them first.  Is there a way you can provide a sample through Google Drive or something similar?

**Reply by jinwoongyoo** [reply]
Hi Scott, Thank you very much for your reply! Let me prepare a subset of my vdc datasets in Google Drive. Would you need only U,V,W, and SMOIS in the data folder along with coordinates folder to test them? Or should I include P and PB folders as well? At present, I have not produced "Elevation" folder in the data folder, which I used to have in the VAPOR 2.x version. Please advise. Thank you. Jinwoong

**Reply by sgpearse** [reply]
Let's start with the original data file.  I'll convert that to VDC and see if I can reproduce what you're seeing.

**Comment by jinwoongyoo** [comment]
Hi Scott,

I made a Google Drive folder to share with you.
Please let me know an email that you want me to share the folder with.
Due to the large file size of the wrfout, I put only two hourly wrfout
files.
Please let me know if you need more than that.
Thank you.
Regards,
Jinwoong

**Reply by sgpearse** [reply]
You can send it to pearse@ucar.edu

**Comment by sgpearse** [comment]
Hey Jinwoong, I just wanted to let you know that I'm still looking into this and haven't forgotten.

**Comment by jinwoongyoo** [comment]
Hi Scott,

Good to hear from you.
Were you able to retrieve some data files you need for your test?
Please let me know if you have any questions or if you need anything.
Thank you.
Regards,
Jinwoong

**Reply by sgpearse** [reply]
Hey, I was able to test your data.  I'm not seeing the vertical stick behavior your describe after importing the wrf files, or after converting them to VDC.  I've attached some screenshots below.  Are you sure your X, Y, and Z field variables are correct?  They should be defaulting to U, V, and W.  I can take a look at a session file that shows what you're seeing if you have one.

Regarding SMOIS, VAPOR's WRF reader does not support the soil_layers vertical coordinate.  This is something that I believe we would consider implementing if you submit [an enhancement request](https://github.com/NCAR/VAPOR/issues/new?assignees=&labels=Enhancement&projects=&template=enhancement-request.md&title=).

![Screen Shot 2024-01-23 at 8 34 49 AM](https://github.com/NCAR/VAPOR/assets/9522770/3e951b91-7ed5-41ad-8dfd-e0ba867bfb23)
![Screen Shot 2024-01-23 at 8 34 57 AM](https://github.com/NCAR/VAPOR/assets/9522770/d4528c03-fb01-4a82-a18f-2a326a6d937d)
